[tayziyi] iP

tayziyi opened the pull request:
### SecretaryW

> If you have SecretaryW, you have everything! 

SecretaryW is your personal assistant, who keeps track of your daily tasks for you! 

Why SecretaryW? 
- Easy to Use! 
- Text-Based
- Actually really fast to learn

All you need to do is,

1. download it from [here](https://github.com/tayziyi/ip/archive/refs/tags/A-Jar.zip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features include:
- [X]  Managing tasks
- [X]  Managing deadlines
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level-0

tayziyi committed:
Delete src/main/java/Duke.class

tayziyi committed:
Delete src/main/java/SecretaryW.class

Commit:
Level-1

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Level-2

tayziyi committed:
Level-3

tayziyi committed:
Level-4

tayziyi committed:
A-TextUiTesting

tayziyi committed:
Level-5

tayziyi committed:
Level-6

tayziyi committed:
A-Enums

tayziyi committed:
Add Save function

ashleygoh1 commented on src/main/java/Task.java:
```diff
@@ -21,6 +21,30 @@ public Task(SecretaryW.TaskType type, String description, String time1) {
+        this.type = type;
+        this.description = description;
+        this.time1 = time1;
+        this.time2 = time2;
```
instead of time1, time2, use more distinguishable name
maybe startTime, endTime?

ashleygoh1 commented on src/main/java/SecretaryW.java:
```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+    private static void checkParts(String[] parts, int length) throws WException {
```
can use a more descriptive variable name
what do parts refer to?

ashleygoh1 commented on src/main/java/SecretaryW.java:
```diff
@@ -0,0 +1,209 @@
+import java.io.*;
```
Imported classes should always be listed explicitly.
example:
import java.io.File;
import java.io.FileWriter;
import java.io.FileReader;

Anant1902 commented on src/main/java/SecretaryW.java:
```diff
@@ -0,0 +1,209 @@
+        Scanner scanner = new Scanner(System.in);
+
+        // ArrayList to store task items
+        ArrayList<Task> taskList = new ArrayList<>();
```
Since this is an array of Tasks, consider changing the taskList to tasks - better practice to use plural form.
```suggestion
        ArrayList<Task> tasks = new ArrayList<>();
```

Anant1902 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,106 @@
+    // Method to convert a Task to a string for writing to the file
+    public String toFileString() {
+        switch (type) {
+            case TODO:
```
Make sure all the cases are on the same indent levels as the switches.

Anant1902 commented on src/main/java/SecretaryW.java:
```diff
@@ -0,0 +1,209 @@
+        TODO, DEADLINE, EVENT
+    }
+
+    private static final String FILE_PATH = "./java/data/SecretaryW.txt";
```
Good job on having all caps for constants

Anant1902 commented on src/main/java/SecretaryW.java:
```diff
@@ -0,0 +1,209 @@
+    private static Task createTaskFromData(String taskData) {
+        String[] parts = taskData.split("\\|");
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1");
```
Good job on starting with 'is' for all boolean variable names

Anant1902 reviewed (commented):
Great job on your code so far. Left some comments!

tayziyi committed:
Add date and time function

tayziyi committed:
Extract Ui, Storage, Parser, TaskList class

tayziyi committed:
Organize into Packages

tayziyi committed:
Stash .txt file

tayziyi committed:
'Merge branch 'add-gradle-support'

tayziyi committed:
Add JUnit Tests

tayziyi committed:
Add JavaDocs

tayziyi committed:
Tweak code to comply with coding standard

tayziyi committed:
Add Find function

tayziyi committed:
Fix conflict for merge branch

tayziyi committed:
Fix conflict for merge branch

tayziyi committed:
Add GUI function

tayziyi committed:
Complete GUI functions

tayziyi committed:
Debug fixed related to exception handling

tayziyi committed:
Changes made to StorageTest.java

tayziyi committed:
Add assert statements to Parser and TaskList class

These assert statements provide additional clarity and help catch
potential issues during development and testing.

Changes Made:
- Included assert statements in other methods to validate method
  arguments and index bounds.

The assert statements were added to the start of the method
to catch potential issues.

tayziyi committed:
Improve code quality for clarity and consistency

Some class contains methods with inconsistent naming and formatting,
making the code less readable and maintainable.

Refactor the classes to use consistent naming conventions and formatting
, improving code clarity and maintainability.

Consistent naming and formatting help make the code more understandable
for other developers and ensure consistency throughout the codebase,
leading to easier maintenance and debugging.

tayziyi committed:
Merge pull request #2 from tayziyi/branch-A-Assertions

Add assert statements to Parser and TaskList class

tayziyi committed:
Merge pull request #3 from tayziyi/branch-A-CodeQuality

Improve code quality for clarity and consistency

tayziyi committed:
Add Help function

tayziyi committed:
Fix debug and improved code quality

tayziyi committed:
Trigger rebuild

tayziyi committed:
Add User Guide

tayziyi committed:
Update README.md

tayziyi committed:
Merge branch 'branch-A-UserGuide'

tayziyi committed:
Add final changes

tayziyi committed:
Fix minor bugs and improve on UI
